### PR TITLE
Init/nutils

### PR DESCRIPTION
**Pull request by Scriptkiddi**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Scriptkiddi** [comment]
@SuperSandro2000

File: pkgs/development/python-modules/treelog/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ lib , buildPythonPackage , fetchPypi, typing-extensions }:
+
+buildPythonPackage rec {
+  pname = "treelog";
+  version = "1.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256:0hnivz4p4llky6djxgcsr9r3j4vr46mkjvp0ksybhpx0fsnhdi81";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/treelog/default.nix:9]:
```suggestion
    sha256 = "0hnivz4p4llky6djxgcsr9r3j4vr46mkjvp0ksybhpx0fsnhdi81";
```

File: pkgs/development/python-modules/treelog/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  propagatedBuildInputs = [
+    typing-extensions
+  ];
+
+  meta = with lib; {
+    description = "Logging framework that organizes messages in a tree structure";
+    homepage = "https://github.com/evalf/treelog";
+    license = licenses.mit;
+    maintainers = [ lib.maintainers.Scriptkiddi ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/treelog/default.nix:22]:
```suggestion
    maintainers = [ maintainers.Scriptkiddi ];
```

File: pkgs/development/python-modules/treelog/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256:0hnivz4p4llky6djxgcsr9r3j4vr46mkjvp0ksybhpx0fsnhdi81";
+  };
+
+  pythonImportsCheck = [ "treelog" ];
+
+  propagatedBuildInputs = [
+    typing-extensions
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/treelog/default.nix:16]:
```suggestion
  checkPhase = ''
    ${python.interpreter} -m unittest
  '';
```

File: pkgs/development/python-modules/treelog/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+buildPythonPackage rec {
+  pname = "treelog";
+  version = "1.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256:0hnivz4p4llky6djxgcsr9r3j4vr46mkjvp0ksybhpx0fsnhdi81";
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/treelog/default.nix:11]:
```suggestion

  propagatedBuildInputs = [
    typing-extensions
  ];

```

File: pkgs/development/python-modules/stringly/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+  propagatedBuildInputs = [
+    typing-extensions
+  ];
+
+  meta = with lib; {
+    description = "Stringly: Human Readable Object Serialization";
+    homepage = "https://github.com/evalf/stringly";
+    license = licenses.mit;
+    maintainers = [ lib.maintainers.Scriptkiddi ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/stringly/default.nix:22]:
```suggestion
    maintainers = [ maintainers.Scriptkiddi ];
```

File: pkgs/development/python-modules/stringly/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256:09fi9dgf27v4qi0mwwms7hpwim9qpyalckd66p7nlmfp6c8bzppq";
+  };
+
+  pythonImportsCheck = [ "stringly" ];
+
+  propagatedBuildInputs = [
+    typing-extensions
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/stringly/default.nix:16]:
```suggestion
  propagatedBuildInputs = [
    typing-extensions
  ];
  
  checkPhase = ''
    ${python.interpreter} -m unittest
  '';
  
  pythonImportsCheck = [ "stringly" ];
```

File: pkgs/development/python-modules/nutils/default.nix [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    numpy
+    treelog
+    stringly
+  ];
+
+  meta = with lib; {
+    description = "Numerical Utilities for Finite Element Analysis";
+    homepage = "http://www.nutils.org/en/stable/";
+    license = licenses.mit;
+    maintainers = [ lib.maintainers.Scriptkiddi ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nutils/default.nix:24]:
```suggestion
    maintainers = [ maintainers.Scriptkiddi ];
```

File: pkgs/development/python-modules/nutils/default.nix [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+  propagatedBuildInputs = [
+    numpy
+    treelog
+    stringly
+  ];
+
+  meta = with lib; {
+    description = "Numerical Utilities for Finite Element Analysis";
+    homepage = "http://www.nutils.org/en/stable/";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nutils/default.nix:22]:
```suggestion
    homepage = "https://www.nutils.org/";
```

File: pkgs/development/python-modules/nutils/default.nix [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    sha256 = "sha256:1sw310l2yb8wbcv2qhik8s928zjh2syzz2qxisglbzski9qdw2x6";
+  };
+
+  pythonImportChecks = [ "nutils" ];
+
+  propagatedBuildInputs = [
+    numpy
+    treelog
+    stringly
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nutils/default.nix:18]:
```suggestion
  propagatedBuildInputs = [
    numpy
    treelog
    stringly
  ];
  
  checkPhase = ''
    ${python.interpreter} -m unittest discover -b -t . -s tests
  '';

  pythonImportChecks = [ "nutils" ];
```

**Review by SuperSandro2000 — COMMENTED** [review]
The prefix for pythonPackages is `pythonPackages.`. Ideally you use ``python310Packages.`` to make ofborg run.

**Comment by Scriptkiddi** [comment]
@SuperSandro2000

File: pkgs/development/python-modules/nutils/default.nix [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+  checkPhase = ''
+    ${python.interpreter} -m unittest discover -b -t . -s tests
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nutils/default.nix:21]:
```
ImportError: Start directory is not importable: 'tests'
```

**Comment by Scriptkiddi** [comment]
@SuperSandro2000

**Pull request merged**